include/seqan3/alignment/pairwise/align_pairwise.hpp

```diff
@@ -23,50 +23,60 @@
- *                            before mentioned tuple concept.
+ * \tparam exec_policy_t      The type of the execution policy; std::is_execution_policy_v must evaluate to true.
+ * \tparam sequence_t         The type of sequence pairs (see details for more information on the type constraints).
  * \tparam alignment_config_t The type of the alignment configuration; must be a seqan3::configuration.
- * \param[in] seq             A tuple with two sequences or an input range over such tuples.
+ * \param[in] exec            The execution policy to use. Defaults to std::execution::seq.
+ * \param[in] seq             A tuple with two sequences or a range over such tuples.
  * \param[in] config          The object storing the alignment configuration.
  * \return A seqan3::alignment_range.
  *
  * \details
  *
- * This function computes the pairwise alignment of a sequence pair or a range over sequence pairs.
- * The pairs can be any class template that models the seqan3::TupleLike.
- * In case of calling the alignment with a single tuple, the value is passed as a std::view::single to the range
- * aware interface. In this case the passed class must be std::CopyConstructible and both element types of the tuple
- * must model std::ranges::ViewableRange because it is copied into the std::view::single. The same
- * is true, if the input range is a temporary non-view range.
+ * This function computes the pairwise alignment for the given sequences. During the setup phase the most efficient
+ * implementation is selected depending on the configurations stored in the given seqan3::configuration object.
+ * In addition, the alignments can be computed in parallel if std::execution::par is given as the execution policy.
+ * On default the algorithm uses the sequenced execution (std::execution::seq).
+ *
+ * ### Compute a single alignment
+ *
+ * In some cases if only a single alignment needs to be computed, the two sequences can be passed as a pair.
+ * The pairs can be any class template that models the seqan3::TupleLike concept. The tuple elements must model
+ * std::ranges::ViewableRange and std::CopyConstructible as the tuple is converted into a range using std::view::single
+ * (by copying/moving the sequence pair) and forwarded to the range aware interface.
  * Accordingly, the following example wouldn't compile:
  * ```cpp
  * std::pair p{"ACGTAGC", "AGTACGACG"};
  * auto rng = align_pairwise(p, config);
  * ```
  * You can use std::tie instead as shown in the example below.
  *
+ * ### Compute multiple alignments
+ *
+ * In many cases one needs to compute multiple pairwise alignments. Accordingly, the align_pairwise interface allows
+ * to pass a range over sequence pairs. The alignment algorithm will be configured only once for all submitted
+ * alignments and then computes the alignments sequenced or in parallel depending on the given execution policy.
+ *
```

Comment:
Add:;
```
Since there is always a certain amount of initial setup required, it is advisable to pass many sequences / sequence-pairs to this algorithm instead of repeatedly calling it with a single pair.
```